packages/react-frontend/src/MyApp.jsx

```diff
@@ -1,228 +1,205 @@
+        const index = contacts.findIndex((contact) => contact.id === id)
+        deleteContact(index)
+            .then((promise) => {
+                const updated = contacts.filter((contact, i) => {
+                    return i !== index
+                })
+                setContacts(updated)
+            })
+            .catch((error) => {
+                console.log(error)
+            })
+    }
 
-  useEffect(() => {
-    fetchFavoriteContacts()
-      .then((json) => {
-        if (json) {
-          setFavoriteContacts(sortContactsByFirstName(json["contact_list"]));
-        } else {
-          setContacts(null);
-        }
-      })
-      .catch((error) => {
-        console.error("Failed to fetch favorite contacts:", error);
-      });
-  }, []);
+    const sortContactsByFirstName = (contacts) => {
+        return contacts
+            .slice()
+            .sort((a, b) => a.first_name.localeCompare(b.first_name))
+    }
 
-  return (
-    <div id="page">
-      <Routes>
-        <Route
-          exact
-          path="/"
-          element={
-            <HomePage
-              contactData={contacts}
-              favoriteContactData={favoriteContacts}
-              removeCharacter={removeOneContact}
-            />
-          }
-        />
-        <Route exact path="/contact/:id" element={<Contact />} />
-        <Route exact path="/edit/:id" element={<Edit />} />
-        <Route exact path="/createContact/" element={<CreateContact />} />
-        <Route exact path="/deleteContact/:id" element={<HomePage />} />
-        {<Route path="/login" element={<LoginPage handleSubmit={loginUser} />} />}
-      </Routes>
-    </div>
-  );
+    return (
+        <div id="page">
+            <Routes>
+                <Route
+                    exact
+                    path="/"
+                    element={
+                        <HomePage
+                            contactData={contacts}
+                            favoriteContactData={favoriteContacts}
+                            removeCharacter={removeOneContact}
+                        />
+                    }
+                />
+                <Route exact path="/contact/:id" element={<Contact />} />
+                <Route
+                    exact
+                    path="/edit/:id"
+                    element={<Edit handleSubmit={removeOneContact(id)} />}
+                />
+                <Route
+                    exact
+                    path="/createContact/"
+                    element={<CreateContact handleSubmit={updateList} />}
+                />
+                <Route exact path="/deleteContact/:id" element={<HomePage />} />
+                {
+                    <Route
+                        path="/login"
+                        element={<LoginPage handleSubmit={loginUser} />}
+                    />
+                }
+            </Routes>
+        </div>
+    )
 }
 
-export default MyApp;
+export default MyApp
```

Comment:
No semi Colon on line 205? (is it not necessary?)

Reply:
no because of the prettier conditions I just added